ariadne/src/Ariadne/Wallet/Cardano/Kernel.hs

```diff
@@ -135,6 +142,21 @@ initPassiveWallet logMessage db = do
 init :: PassiveWallet -> IO ()
 init PassiveWallet{..} = _walletLogMessage Info "Passive Wallet kernel initialized"
 
+createApplyHandler :: PassiveWallet -> (OldestFirst NE Blund -> IO ())
+createApplyHandler pw = applyBlocks pw . mapMaybe' blundToResolvedBlock
+  where
+    mapMaybe' :: (a -> Maybe b) -> OldestFirst NE a -> OldestFirst [] b
+    mapMaybe' f = OldestFirst . mapMaybe f . toList . getOldestFirst
+    rightToJust   = either (const Nothing) Just
+    -- The use of the unsafe constructor 'UnsafeRawResolvedBlock' is justified
+    -- by the invariants established in the 'Blund'.
+    blundToResolvedBlock :: Blund -> Maybe ResolvedBlock
+    blundToResolvedBlock (b,u)
+        = rightToJust b <&> \mainBlock ->
+            fromRawResolvedBlock
+            $ UnsafeRawResolvedBlock mainBlock (map (map fromJust) $ undoTx u)
```

Comment:
I don't like that you inlined stuff from `Cardano.Wallet.WalletLayer.Kernel` here. It increases diff with IOHK. I find it better to copy-paste `Cardano.Wallet.WalletLayer.Kernel` into our code (and some other modules probably) and use `Cardano.Wallet.WalletLayer.Kernel.bracketPassiveWallet`.